scripts/validate_string_concatenation.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+"""
+GH #30454
+
+Check where there is a string that needs to be concatenated.
+
+This is necessary after black formating,
+where for example black transforms this:
+
+>>> foo = (
+...         "bar "
+...         "baz"
+...     )
+
+into this:
+
+>>> foo = ("bar " "baz")
+
+Black is not considering this as an
+issue (see https://github.com/psf/black/issues/1051), so we are checking
+it here.
+"""
+
+import os
+import sys
+import token
+import tokenize
+
+FILE_EXTENSIONS_TO_CHECK = [".pxd", ".py", ".pyx", ".pyx.ini"]
+
+
+def main():
+    path = sys.argv[1]
+
+    if not os.path.exists(path):
+        raise ValueError("Please enter a valid path, to a file/directory.")
+
+    if os.path.isfile(path):
+        # Means that the given path is of a single file.
+        sys.exit(is_concatenated(path))
+
+    status_codes = set()
+    # Means that the given path is of a directory.
+    for subdir, _, files in os.walk(path):
+        for file_name in files:
+            ext = full_ext(os.path.join(subdir, file_name))
```

Comment:
You can simply use `if any(file_name.endswith(ext) for ext in [".pxd", ".py", ".pyx", ".pyx.ini"]):` and get rid of the `full_ext` function.